validation.py

```diff
@@ -89,3 +91,30 @@ def validate(self, value):
                     msg = f"Expected {value!r} to be a normalised string."
                     logging.error(msg)
                     raise TypeError(msg)
+
+class Date(Validator):
+    def __init__(self, earliest_date=None, latest_date=None, first_of_month=False):
+        self.earliest_date = earliest_date
+        self.latest_date = latest_date
+        self.first_of_month = first_of_month
+
+    def validate(self, value):
+        if not isinstance(value, (dt.date)):
+            msg = f'Expected {value!r} to be an int or float'
+            logging.error(msg)
+            raise TypeError(msg)
+        if self.earliest_date is not None and value < self.earliest_date:
+            msg = f'Expected {value!r} to be at least {self.earliest_date!r}'

```

Comment:
Couple things. Try use double quotes for all your strings, and I think putting '!r' at the end of the variable will break the message.